Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -902,13 +902,20 @@ void NetworkConnectionToWebProcess::domCookiesForHost(const URL& url, Completion
 
 #if HAVE(COOKIE_CHANGE_LISTENER_API)
 
-void NetworkConnectionToWebProcess::subscribeToCookieChangeNotifications(const String& host)
+void NetworkConnectionToWebProcess::subscribeToCookieChangeNotifications(const URL& url, const URL& firstParty, WebCore::FrameIdentifier frameID, WebCore::PageIdentifier pageID, WebCore::ShouldRelaxThirdPartyCookieBlocking shouldRelaxThirdPartyCookieBlocking, CompletionHandler<void(bool)>&& completionHandler)
 {
+    MESSAGE_CHECK_COMPLETION(protectedNetworkProcess()->allowsFirstPartyForCookies(m_webProcessIdentifier, firstParty), completionHandler(false));
+
+    auto host = url.host().toString();
     ASSERT(!m_hostsWithCookieListeners.contains(host));
```

Comment:
My understanding is you need to drop this assertion because web process could send request for the same host, but with different url / firstParty / frameID / pageID / shouldRelaxThirdPartyCookieBlocking with latest impl